controller/action/DownloadActions.class.php

```diff
@@ -28,6 +28,29 @@ public static function executeGetDaemonRedirect(string $os)
 
   public static function executeGet()
   {
+    if (static::prepareDownloadInformation()) {
+      return ['download/get', static::prepareDownloadInformation()];
+    }
+
+    return ['download/get'];
+  }
+
+  public static function prepareListPartial(array $vars)
+  {
+    return $vars + ['osChoices' => isset($vars['excludeOs']) ?
+      array_diff_key(OS::getAll(), [$vars['excludeOs'] => null]) :
+      OS::getAll()
+    ];
+  }
+
+  public static function prepareButtonPartial() {
+    if (static::prepareDownloadInformation()) {
+      return static::prepareDownloadInformation();
+    }
+    return [];
+  }
+
+  private static function prepareDownloadInformation() {
```

Comment:
`protected` over `private` generally. But in the case of `prepareXXX` functions, they should be `public`.

`prepareFoo` will be called automatically when including `_foo.php` as a view.